### ISSUE TITLE
Negative Loss values

### ISSUE THREAD

**Issue by max-se3** [yd-yin/RotationLaplace#4]
Hi, 
thanks for the great work! Really appreciate your paper. 
When trying to integrate the Laplace approach of your paper into my 3D object detection model I ran into some issues.

I got it so far implemented as that the approach can learn rotations really well on an overfit dataset.
On the full dataset this doesn't work as well yet which is probably due to the NLL_loss often producting negative loss values.

Based on the formulas from your paper this shouldn't be possible but when I test the NLL_loss with the laplace approach on the initial 9d, raw, unconstrained values I got from the network, this often results in there being a negative loss.
I've already tried fixing this by ensuring type consistency, etc. but nothing seems to work so far.

Am I missing something here or do you have any advice on how I could fix this?

One set of examples values I tested were these

raw_pred = torch.tensor([
       [ -4.2500,   1.0703,  -5.9375],
        [ -0.7988,   0.6445,  -8.7031],
        [ -3.0820,  -1.2852, -21.1562]
    ]).unsqueeze(0)

    target = torch.tensor([
       [ 0.8829, -0.4386, -0.1675],
        [-0.4684, -0.7984, -0.3783],
        [ 0.0321,  0.4124, -0.9104]
    ]).unsqueeze(0)
    
    grids = torch.from_numpy(np.load(grids_path))#.to(dtype=torch.float32)

    losses, pred_orth = NLL_loss('RLaplace', raw_pred, target, grids)

and for these the loss becomes ~ -0.38

Thanks a lot in advance!

**Comment by yd-yin** [comment]
Hi, thanks for your interest!

Actually I don't think a negative loss is something wrong. For NLL loss, when pdf > 1, -log(pdf) would be negative.

For deep learning, a negative loss is also fine. The goal of the optimization is making the loss smaller, no matter it is positive or negative.

This is not a special property for our distribution.
